node/modules/lp2p/pubsub.go

```diff
@@ -383,36 +374,12 @@ func GossipSub(in GossipIn) (service *pubsub.PubSub, err error) {
 
 	options = append(options, pubsub.WithPeerGater(pgParams))
 
-	allowTopics := []string{
+	allowTopics := append(in.Tp,
 		build.BlocksTopic(in.Nn),
 		build.MessagesTopic(in.Nn),
 		build.IndexerIngestTopic(in.Nn),
-	}
-
-	if build.IsF3Enabled() {
-		f3TopicName := manifest.PubSubTopicFromNetworkName(gpbft.NetworkName(in.Nn))
-		allowTopics = append(allowTopics, f3TopicName)
-
-		// allow dynamic manifest topic and the new topic names after a reconfiguration.
-		// Note: This is pretty ugly, but I tried to use a regex subscription filter
-		// as the commented code below, but unfortunately it overwrites previous filters. A simple fix would
-		// be to allow combining several topic filters, but for now this works.
-		//
-		// 	pattern := fmt.Sprintf(`^\/f3\/%s\/0\.0\.1\/?[0-9]*$`, in.Nn)
-		// 	rx, err := regexp.Compile(pattern)
-		// 	if err != nil {
-		// 		return nil, xerrors.Errorf("failed to compile manifest topic regex: %w", err)
-		// 	}
-		// 	options = append(options,
-		// 		pubsub.WithSubscriptionFilter(
-		// 			pubsub.WrapLimitSubscriptionFilter(
-		// 				pubsub.NewRegexpSubscriptionFilter(rx),
-		// 				100)))
-		allowTopics = append(allowTopics, manifest.ManifestPubSubTopicName)
-		for i := 0; i < MaxDynamicManifestChangesAllowed; i++ {
-			allowTopics = append(allowTopics, f3TopicName+"/"+fmt.Sprintf("%d", i))
-		}
-	}
+		build.F3TopicName(in.Nn),
```

Comment:
Always enabled. The filter exists to protect against a DoS attack with too many topic names, adding a single one won't make a difference.